superset/assets/javascripts/explorev2/components/ChartContainer.jsx

```diff
@@ -37,125 +33,80 @@ const propTypes = {
-    this.state.viz.render();
-  }
-
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.chartStatus === 'loading') {
-      const mockSlice = this.getMockedSliceObject(nextProps);
-      this.setState({
-        mockSlice,
-        viz: visMap[nextProps.viz_type](mockSlice),
-      });
+    try {
+      visMap[this.props.viz_type](mockSlice, this.props.queryResponse);
+      this.setState({ mockSlice });
+    } catch (e) {
+      this.props.actions.chartRenderingFailed(e);
     }
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.chartStatus === 'loading') {
-      this.state.viz.render();
-    }
-    if (prevProps.height !== this.props.height) {
-      this.state.viz.resize();
+    if (
+        prevProps.queryResponse !== this.props.queryResponse ||
+        prevProps.height !== this.props.height
+      ) {
+      this.renderViz();
     }
   }
 
   getMockedSliceObject(props) {
     return {
       viewSqlQuery: props.query,
-
-      data: {
-        csv_endpoint: props.csv_endpoint,
-        json_endpoint: props.json_endpoint,
-        standalone_endpoint: props.standalone_endpoint,
-      },
-
       containerId: props.containerId,
-
-      jsonEndpoint: () => props.json_endpoint,
-
       container: {
         html: (data) => {
           // this should be a callback to clear the contents of the slice container
           $(this.state.selector).html(data);
         },
-
         css: (dim, size) => {
           // dimension can be 'height'
           // pixel string can be '300px'
           // should call callback to adjust height of chart
           $(this.state.selector).css(dim, size);
         },
         height: () => parseInt(this.props.height, 10) - 100,
-
-        show: () => { this.render(); },
-
+        show: () => { },
         get: (n) => ($(this.state.selector).get(n)),
-
         find: (classname) => ($(this.state.selector).find(classname)),
-
       },
-
       width: () => this.chartContainerRef.getBoundingClientRect().width,
-
       height: () => parseInt(this.props.height, 10) - 100,
-
       selector: this.state.selector,
-
       setFilter: () => {
         // set filter according to data in store
         // used in FilterBox.onChange()
       },
-
       getFilters: () => (
         // return filter objects from viz.formData
         {}
       ),
-
-      done: (payload) => {
-        // finished rendering callback
-        // Todo: end timer and chartLoading set to success
-        props.actions.chartUpdateSucceeded(payload.query);
-      },
-
+      done: () => {},
       clearError: () => {
         // no need to do anything here since Alert is closable
         // query button will also remove Alert
       },
-
-      error(msg) {
-        let payload = { error: msg };
-        try {
-          payload = JSON.parse(msg);
-        } catch (e) {
-          // pass
-        }
-        props.actions.chartUpdateFailed(payload.error, payload.query);
-      },
-
+      error() {},
```

Comment:
Alright I can change this. I wish the linter would enforce this so we wouldn't have to go back and forth on it.